src/Model/Payment/AbstractPayment.php

```diff
@@ -0,0 +1,170 @@
+     */
+    const XML_CONFIG_STATUS_NOT_PAID = '';
+    /**
+     * The name of the module, as defined in etc/module.xml
+     */
+    const MODULE_NAME = '';
+    /**
+     * The code of the magento payment method
+     */
+    const PAYMENT_CODE = '';
+
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+    /** @var Manager */
+    private $moduleManager;
+    /** @var PaymentHelper */
+    private $paymentHelper;
+    /** @var Utility */
+    private $utility;
+
+    /**
+     * @param ScopeConfigInterface $scopeConfig
+     * @param Manager              $moduleManager
+     * @param PaymentHelper        $paymentHelper
+     * @param Utility              $utility
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        Manager $moduleManager,
+        PaymentHelper $paymentHelper,
+        Utility $utility
+    ) {
+        $this->scopeConfig   = $scopeConfig;
+        $this->moduleManager = $moduleManager;
+        $this->paymentHelper = $paymentHelper;
+        $this->utility       = $utility;
+    }
+
+    /**
+     * Runs initial setup functions
+     */
+    public function setUp(): void
+    {
+    }
+
+    /**
+     * Returns the concrete payment model instance
+     *
+     * @return MethodInterface|null
+     * @throws LocalizedException
+     */
+    public function getPaymentModel(): ?MethodInterface
+    {
+        return $this->isValid() ? $this->paymentHelper->getMethodInstance(static::PAYMENT_CODE) : null;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isValid(): bool
+    {
+        return $this->isModuleActive() && $this->isEnabled();
+    }
+
+    /**
+     * Check if the payment module is enabled
+     *
+     * @return bool
+     */
+    public function isModuleActive(): bool
+    {
+        return $this->moduleManager->isEnabled(static::MODULE_NAME);
+    }
+
+    /**
+     * Checks if a payment method is enabled
+     *
+     * @return bool
+     */
+    public function isEnabled(): bool
+    {
+        $this->scopeConfig->isSetFlag(static::XML_CONFIG_ENABLED, ScopeInterface::SCOPE_STORE);
+    }
+
+    /**
+     * Allows manipulation of order data
+     *
+     * @param MagentoOrder  $magentoOrder
+     * @param ShopgateOrder $shopgateOrder
+     */
+    public function manipulateOrderWithPaymentData(MagentoOrder $magentoOrder, ShopgateOrder $shopgateOrder): void
+    {
+    }
+
+    /**
+     * Sets order status based on configuration
+     *
+     * @param MagentoOrder  $magentoOrder
+     * @param ShopgateOrder $shopgateOrder
+     *
+     * @throws LocalizedException
+     */
+    public function setOrderStatus(MagentoOrder $magentoOrder, ShopgateOrder $shopgateOrder): void
+    {
+        $orderStatusConfig = $shopgateOrder->getIsPaid()
+            ? static::XML_CONFIG_STATUS_PAID
+            : static::XML_CONFIG_STATUS_NOT_PAID;
+        $orderStatus = $this->scopeConfig->getValue($orderStatusConfig, ScopeInterface::SCOPE_STORE);
```

Comment:
is there are store number that has t o go here? ya know, in case it is importing it to a store specific thing with a store specific config

Reply:
I switched the way, we are doing it in the export module using our own config loader.